tests/jenkins/TestUploadToS3.groovy

```diff
@@ -0,0 +1,73 @@
+import jenkins.tests.BuildPipelineTest
+import org.junit.Before
+import org.junit.Test
+
+import static org.hamcrest.CoreMatchers.notNullValue
+import static org.hamcrest.MatcherAssert.assertThat
+
+class TestUploadToS3 extends BuildPipelineTest {
+
+    static void setUpVariables(binding, helper){
+        binding.setVariable('ARTIFACT_UPLOAD_ROLE_NAME', 'Dummy_Upload_Role')
+        binding.setVariable('AWS_ACCOUNT_PUBLIC', 'dummy_account')
+        binding.setVariable('ARTIFACT_BUCKET_NAME', 'dummy_bucket_name')
+        helper.registerAllowedMethod("s3Upload", [Map])
+        helper.registerAllowedMethod("withAWS", [Map, Closure], { args, closure ->
+            closure.delegate = delegate
+            return helper.callClosure(closure)
+        })
+    }
+
+    @Before
+    void setUp() {
+        super.setUp()
+    }
+
+    @Test
+    void testSignArtifacts() {
+
+        def sourcePath = '/tmp/src/path'
+        def bucket = 'dummy_bucket'
+        def path = '/upload/path'
+
+        binding.setVariable('sourcePath', sourcePath)
+        binding.setVariable('bucket', bucket)
+        binding.setVariable('path', path)
+
+        setUpVariables(binding, helper)
+
+        super.testPipeline("tests/jenkins/jobs/UploadToS3_Jenkinsfile")
+
+        verifyUploadToS3Params(helper, sourcePath, bucket, path)
+    }
+
+    static void verifyUploadToS3Params(helper, sourcePath, bucket, path) {
+        assert helper.callStack.findAll { call ->
+            call.methodName == 'uploadToS3'
+        }.size() > 0
+
+        helper.callStack.findAll { call ->
+            call.methodName == 'uploadToS3'
+        }.each { call ->
```

Comment:
This is going to verify that all calls to `uploadToS3` are the same. So it won't extend if a job calls `uploadToS3` twice.

But, it can be improved when needed.

Reply:
I don't understand this completely. It should, since the params need to follow the constraints for all calls, and the values need to match for any one of the calls as checked here -

https://github.com/opensearch-project/opensearch-build/pull/1471/files/2e5c8f5ad2a5161db2e81ab30c28d9507502ff8b#diff-54798da6d4691a9cbd704cf6e4af0db9a6ae906652a7c3bc1bb494d7111280ecR57-R71

Am I missing something?

Reply:
Sorry, my mistake here. Yes, I believe that this will work.